src/test/java/com/hillel/items_exchange/controller/AuthControllerIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+import org.junit.jupiter.api.Test;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@DBRider
+@AutoConfigureMockMvc
+@Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:index-reset.sql")
+public class AuthControllerIntegrationTest extends AuthControllerIntegrationTestUtil {
+
+    private final MockMvc mockMvc;
+
+    @Autowired
+    public AuthControllerIntegrationTest(MockMvc mockMvc) {
+        super();
+        this.mockMvc = mockMvc;
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_shouldCreateNewUserAndReturnCreated() throws Exception {
+        UserRegistrationDto validUserRegistrationDto = createValidUserRegistrationDto();
+        mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(validUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isCreated());
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUserRegistrationDtoIsEmpty_shouldReturnBadRequest() throws Exception {
+
+        mockMvc.perform(post(REGISTER_URL, "")
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUserRegistrationDtoIsNull_shouldReturnBadRequest() throws Exception {
+
+        mockMvc.perform(post(REGISTER_URL, (Object) null)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUserEmailExists_shouldReturnUnprocessableEntityAndThrowUnprocessableEntityException()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithExistentEmail();
+        MvcResult result = this.mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isUnprocessableEntity())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(UnprocessableEntityException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUsernameExists_shouldReturnUnprocessableEntityAndThrowUnprocessableEntityException()
```

Comment:
When you create new one method at the dao layer that will check if the user exists by email or login you won't need this test anymore.

Reply:
Got it